Source/Core/GroundPolylineGeometry.js

```diff
@@ -0,0 +1,1054 @@
+        Cartographic,
+        Check,
+        ComponentDatatype,
+        DeveloperError,
+        CesiumMath,
+        defaultValue,
+        defined,
+        defineProperties,
+        Ellipsoid,
+        EllipsoidGeodesic,
+        EncodedCartesian3,
+        GeographicProjection,
+        Geometry,
+        GeometryAttribute,
+        IntersectionTests,
+        Matrix3,
+        Plane,
+        Quaternion,
+        Rectangle,
+        WebMercatorProjection) {
+    'use strict';
+
+    var PROJECTIONS = [GeographicProjection, WebMercatorProjection];
+    var PROJECTION_COUNT = PROJECTIONS.length;
+
+    var MITER_BREAK_SMALL = Math.cos(CesiumMath.toRadians(30));
+    var MITER_BREAK_LARGE = Math.cos(CesiumMath.toRadians(150));
+
+    // Initial heights for constructing the wall.
+    // Keeping WALL_INITIAL_MIN_HEIGHT near the ellipsoid surface helps
+    // prevent precision problems with planes in the shader.
+    // Putting the start point of a plane at ApproximateTerrainHeights._defaultMinTerrainHeight,
+    // which is a highly conservative bound, usually puts the plane origin several thousands
+    // of meters away from the actual terrain, causing floating point problems when checking
+    // fragments on terrain against the plane.
+    // Ellipsoid height is generally much closer.
+    // The initial max height is arbitrary.
+    // Both heights are corrected using ApproximateTerrainHeights for computing the actual volume geometry.
+    var WALL_INITIAL_MIN_HEIGHT = 0.0;
+    var WALL_INITIAL_MAX_HEIGHT = 1000.0;
+
+    /**
+     * A description of a polyline on terrain. Only to be used with GroundPolylinePrimitive.
+     *
+     * @alias GroundPolylineGeometry
+     * @constructor
+     *
+     * @param {Object} [options] Options with the following properties:
+     * @param {Number} [options.width=1.0] The screen space width in pixels.
+     * @param {Cartesian3[]} [options.positions] An array of {@link Cartesian3} defining the polyline's points. Heights above the ellipsoid will be ignored.
+     * @param {Number} [options.granularity=9999.0] The distance interval used for interpolating options.points. Defaults to 9999.0 meters. Zero indicates no interpolation.
+     * @param {Boolean} [options.loop=false] Whether during geometry creation a line segment will be added between the last and first line positions to make this Polyline a loop.
+     * @param {Ellipsoid} [options.ellipsoid=Ellipsoid.WGS84] Ellipsoid that input positions will be clamped to.
+     * @param {MapProjection} [options.projection] Map Projection for projecting coordinates to 2D.
+     *
+     * @see GroundPolylinePrimitive
+     */
+    function GroundPolylineGeometry(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * The screen space width in pixels.
+         * @type {Number}
+         */
+        this.width = defaultValue(options.width, 1.0); // Doesn't get packed, not necessary for computing geometry.
```

Comment:
This is imitating what's in `Polyline.js` and `PolylineGeometry.js`.
`GroundPolylinePrimitive` uses this to auto-generate instance attributes when they aren't otherwise specified.